Orange/widgets/classify/tests/test_owsvmclassification.py

```diff
@@ -1,107 +1,69 @@
+        self.parameters = [
+            ParameterMapping("C", self.widget.c_spin),
+            ParameterMapping("gamma", self.widget._kernel_params[0],
+                             values=values, setter=setter, getter=getter),
+            ParameterMapping("coef0", self.widget._kernel_params[1]),
+            ParameterMapping("degree", self.widget._kernel_params[2]),
+            ParameterMapping("tol", self.widget.tol_spin),
+            ParameterMapping("max_iter", self.widget.max_iter_spin[1])]
 
-    def test_optimization_box_checked(self):
-        """ Check if spinner box for iteration limit is enabled or disabled """
-        for value in (True, False):
-            self.widget.max_iter_check_box.setChecked(value)
-            self.assertEqual(self.widget.max_iter_check_box.isChecked(), value)
-            self.assertEqual(self.spin_boxes[6].isEnabled(), value)
+    def test_parameters_unchecked(self):
+        """Check learner and model for various values of all parameters
+        when Iteration limit is not checked
+        """
+        self.widget.max_iter_spin[0].setCheckState(False)
+        self.parameters[-1] = DefaultParameterMapping("max_iter", -1)
+        self.test_parameters()
 
-    def test_type_button_checked(self):
-        """ Check if SVM type is selected after click """
-        self.widget.type_box.buttons[0].click()
-        self.assertEqual(self.widget.type_box.buttons[0].isChecked(), True)
-        self.widget.type_box.buttons[1].click()
-        self.assertEqual(self.widget.type_box.buttons[1].isChecked(), True)
+    def test_parameters_svm_type(self):
+        """Check learner and model for various values of all parameters
+        when NuSVM is chosen
+        """
+        self.assertEqual(self.widget.svmtype, OWSVMClassification.C_SVC)
+        # setChecked(True) does not trigger callback event
+        self.widget.nu_radio.click()
+        self.assertEqual(self.widget.svmtype, OWSVMClassification.Nu_SVC)
+        self.parameters[0] = ParameterMapping("nu", self.widget.nu_spin)
+        self.test_parameters()
 
-    def test_type_button_properties_visible(self):
-        """ Check if spinner box in SVM type are visible """
-        self.assertEqual(not self.spin_boxes[0].isHidden(), True)
-        self.assertEqual(not self.spin_boxes[1].isHidden(), True)
-
-    def test_data_before_apply(self):
-        """ Check if data are set """
-        self.widget.set_data(Table("iris")[:100])
-        self.widget.apply()
-        self.assertEqual(len(self.widget.data), 100)
-
-    def test_output_signal_learner(self):
-        """ Check if we have on output learner """
-        self.widget.kernel_box.buttons[0].click()
-        self.widget.set_data(Table("iris")[:100])
-        self.widget.apply()
-        self.assertNotEqual(self.widget.learner, None)
+    def test_kernel_equation(self):
+        """Check if the right equation is written according to kernel """
+        for i in range(4):
```

Comment:
The old class included a test case that checked, whether correct combination of kernel selection, parameters and formula is used when widget is shown. (This was a regression test for bug we have encountered before.) Could the same check be added to this test (and the next one for parameter box visibiity)?